### PR TITLE
Lfa 1122 integrate ch sdk node project

### PR DESCRIPTION
**Pull request by mwestacott**
Won't merge until Statler/Toro1 env is working again.

### CODE REVIEW

File: src/client/apiclient.ts [hunk 1]
```diff
@@ -118,37 +96,46 @@ const getApiData = async (config: AxiosRequestConfig): Promise<any> => {
...
+  if (sdkResponse.httpStatusCode >= 400) {
+    throw {
+      status: sdkResponse.httpStatusCode,
+    };
+  }
+
+  logger.debug("Data from company profile SDK call " + JSON.stringify(sdkResponse, null, 2));
 
-  const data = await getApiData(config) as CompanyProfileResource;
+  const companyProfile: CompanyProfile = sdkResponse.resource as CompanyProfile;
```
Comment (kevbradwick) [inline src/client/apiclient.ts]: no need to add type to the `companyProfile` variable as you are casting it explicitly to `CompanyProfile`. you can simple omit it e.g. `const companyProfile = sdkResponse.resource as CompanyProfile`

File: src/test/client/apiclient.spec.unit.ts [hunk 2]
```diff
@@ -1,70 +1,86 @@
...
-      address_line_2: "line2",
-      postal_code: "post code"
-    }
-  },
-  status: 200,
-  statusText: "OK",
-  headers: "header",
-  config: {}
-};
+const mockCreateApiClient = (<unknown>createApiClient as jest.Mock<typeof createApiClient>);
```
Comment (kevbradwick) [inline src/test/client/apiclient.spec.unit.ts]: the correct cast for this is a `MockedFunction` as that is was `createApiClient` is.

```typescript
const mockCreateApiClient = (createApiClient as jest.MockedFunction<typeof createApiClient>);

mockCreateApiClient.mockImplementation(() => {
  return api;
});```

Comment (kevbradwick) [reply src/test/client/apiclient.spec.unit.ts]: also, the mocking can be simplified to only mock the `CompanyProfileService`. for example;

```typescript
import {getCompanyProfile, ExtensionsCompanyProfile} from "../../client/apiclient";
import * as mockUtils from "../mock.utils";
import Resource from "ch-sdk-node/dist/services/resource";
import {CompanyProfile} from "ch-sdk-node/dist/services/company-profile";
import CompanyProfileService from "ch-sdk-node/dist/services/company-profile/service";

//////////////////
//Set up mocks
jest.mock("ch-sdk-node/dist/services/company-profile/service");

// end of set up mocks
///////////////////////

describe("apiclient company profile unit tests", () => {

  const mockGetCompanyProfile = (CompanyProfileService.prototype.getCompanyProfile as jest.Mock);

  beforeEach(() => {
    mockGetCompanyProfile.mockReset();
  });

  it("converts company number to uppercase", async () => {
    mockGetCompanyProfile.mockResolvedValueOnce(dummySDKResponse);
    const company = await getCompanyProfile("sc100079", mockUtils.ACCESS_TOKEN);
    expect(company.incorporationDate).toEqual("26 June 1872");
    const args = mockGetCompanyProfile.mock.calls[0][0];
    expect(args).toContain("SC100079");
  });

  it("returns an ExtensionsCompanyProfile object", async () => {
    mockGetCompanyProfile.mockResolvedValueOnce(dummySDKResponse);
    const company = await getCompanyProfile("00006400", mockUtils.ACCESS_TOKEN);
    expect(company).toEqual(expectedProfile);
  });
});```

Comment (mwestacott) [reply src/test/client/apiclient.spec.unit.ts]: Agreed. Fixing both of these now.

File: src/test/client/apiclient.spec.unit.ts [hunk 3]
```diff
@@ -1,70 +1,74 @@
...
+    mockGetCompanyProfile.mockReset();
+  });
 
   it("converts company number to uppercase", async () => {
+    mockGetCompanyProfile.mockResolvedValueOnce(dummySDKResponse);
     const company = await getCompanyProfile("sc100079", mockUtils.ACCESS_TOKEN);
     expect(company.incorporationDate).toEqual("26 June 1872");
-    const args = mockRequest.mock.calls[0][0];
-    expect(args.url).toContain("SC100079");
+    const args = mockGetCompanyProfile.mock.calls[0][0];
```
Comment (kevbradwick) [inline src/test/client/apiclient.spec.unit.ts]: can we change this to use `expect().toBeCalledWith()`?

```typescript
    expect(mockGetCompanyProfile).toBeCalledWith("SC100079");
```

it describes the assertion better than accessing the param via array indexes.

**Pull request merged**